### DISCUSSION TITLE
Dataframe visualization looks bugged when trying to display more than two correlations

### DISCUSSION THREAD

**Discussion by f-fuchs** [marimo-team/marimo#2655] [Q&A] [answered]
When I try to display a DataFrame with more than two correlations I get the following output:
![image](https://github.com/user-attachments/assets/fcac8b62-f6b4-41e8-88c1-97e38e418cb2)
![image](https://github.com/user-attachments/assets/db6caf81-9e81-493a-aa57-3f12dc2c50c1)

Am I doing something wrong or is this a bug?

**Comment by f-fuchs** [comment]
this also occurs when casting the DataFrame from polars to pandas first or wrapping it in mo.ui.table.

this is how it looks combined with a normal column
![image](https://github.com/user-attachments/assets/bcde0fb2-5d7a-41b0-9bd3-e2227b572df1)

**Reply by mscolnick** [reply]
@f-fuchs - is it the wide-columns? or what specifically looks bugged?

**Reply by f-fuchs** [reply]
if we look at the first picture for example. the dataframe has 3 columns but 4 headings, because the last one is duplicated. once without the green bar above the actual column and a second time right of the column with the green bar

![tmp](https://github.com/user-attachments/assets/77815682-f0d1-4af4-8bfa-2e3a57f2c5b1)

**Reply by mscolnick** [reply]
If there is any reproduction you can share, we can maybe take a look

**Reply by f-fuchs** [reply]
```
# ruff: noqa

import marimo

__generated_with = "0.9.10"
app = marimo.App(width="full")


@app.cell
def __():
    import polars as pl
    return (pl,)


@app.cell
def __(pl):
    df = pl.scan_csv("hf://datasets/nameexhaustion/polars-docs/iris.csv").collect()
    return (df,)


@app.cell
def __(df):
    df
    return


@app.cell
def __(df):
    df.columns
    return


@app.cell
def __(df, pl):
    df.select(
        pl.corr("sepal_length", "sepal_width", method="spearman").alias(
            "corr(sepal_length,sepal_width)"
        ),
        pl.corr("sepal_length", "petal_length", method="spearman").alias(
            "corr(sepal_length,petal_length)"
        ),
        pl.corr("sepal_length", "petal_width", method="spearman").alias(
            "corr(sepal_length,petal_width)"
        ),
    )
    return


@app.cell
def __():
    return


if __name__ == "__main__":
    app.run()

```
above code results in following output for me:
![image](https://github.com/user-attachments/assets/5507f680-b070-41b9-bf3e-29bf5b1a1241)

as you can see I have three columns but 5 headers.

**Reply by mscolnick** [reply, marked as answer]
This might be fixed in the latest release (not sure what changed) - can you try the latest marimo?

![image](https://github.com/user-attachments/assets/b08cb1ac-7553-4b9a-acd6-426d541a7916)

**Reply by f-fuchs** [reply]
thx, its fixed in the latest release 👍
